sklearn/cluster/_kmeans.py

```diff
@@ -1944,3 +1957,37 @@ def _more_tags(self):
                 'zero sample_weight is not equivalent to removing samples',
             }
         }
+
+
+def kmeans_plusplus(X, n_clusters, x_squared_norms=None,
+                    random_state=None, n_local_trials=None):
+    """Add some documentation here
+
+    """
+
+    # Check parameters
+    if x_squared_norms is None:
+        x_squared_norms = row_norms(X, squared=True)
+    elif len(x_squared_norms) != X.shape[0]:
+        warnings.warn(
+            "x_squared_norms should be of length n_samples. "
+            "computing default norms"
+                     )
+        x_squared_norms = row_norms(X, squared=True)
+
+    random_state = check_random_state(random_state)
+
+    # Check data
+    check_array(X, accept_sparse='csr',
+                dtype=[np.float64, np.float32], order='C')
+
+    # Call private k-means++
+    centers, indices = _k_init(X, n_clusters, x_squared_norms,
+                               random_state, n_local_trials)
+
+    if sp.issparse(centers):
+        centers = centers.toarray()
+
+    _validate_center_shape(X, n_clusters, centers)
```

Comment:
This is not necessary, if an error is raised by `validate_center_shape`, it means the function did not work as intended. This should be handled by good tests instead